xarray/tests/test_variable.py

```diff
@@ -139,6 +139,15 @@ def _assertIndexedLikeNDArray(self, variable, expected_value0,
         # test identity
         assert variable.equals(variable.copy())
         assert variable.identical(variable.copy())
+        # test that we obtain a view, not a copy, when taking a 0d slice
+        if (
+                isinstance(variable._data, np.ndarray)
+                and isinstance(variable[0]._data, np.ndarray)
+        ):
+            assert (
+                variable[0]._data.__array_interface__["data"][0]
+                == variable._data.__array_interface__["data"][0]
+            )
```

Comment:
Can you add more explicit tests that other developers easily find what we are testing?
e.g. 
```python

a = variable[0]
a = 3
assert a == variable[0]
```

Reply:
It might also be easier just to write a separate dedicated method for this, rather than squeezing it into the existing `_assertIndexedLikeNDArray`.

Reply:
You're right, my thinking about tests was a little muddled. The new solution should be better.